src/Controller/Admin/UserController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Controller\Admin;
+
+use App\Entity\User;
+use App\Form\Type\Admin\CreateNewUserType;
+use App\Form\Type\Admin\UpdateUserType;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+
+class UserController extends AbstractController
+{
+    private UserRepository $userRepository;
+
+    private EntityManagerInterface $entityManager;
+
+    private UserPasswordHasherInterface $userPasswordHasher;
+
+
+    public function __construct(
+        UserRepository $userRepository,
+        EntityManagerInterface $entityManager,
+        UserPasswordHasherInterface $userPasswordHasher
+    ) {
+        $this->userRepository = $userRepository;
+        $this->entityManager = $entityManager;
+        $this->userPasswordHasher = $userPasswordHasher;
+    }
+
+    /**
+     * @Route("admin/users", name="admin_users", methods={"GET"})
+     */
+    public function getAllUsers(Request $request): Response
+    {
+        $users = $this->userRepository->findAll();
+
+        return $this->render('Admin/allUsers.html.twig', [
+            'users' => $users
+        ]);
+    }
+
+    /**
+     * @Route("admin/user/{id}", name="user_update", methods={"GET", "POST"}, requirements={"id"="\d+"})
+     */
+    public function updateUser(Request $request, int $id): Response
+    {
+        $user = $this->userRepository->findOneBy(['id' => $id]);
+        if (null === $user) {
+            $this->addFlash(
+                'error',
+                'User does not exist!'
+            );
+        }
+
+        $form = $this->createForm(UpdateUserType::class)->handleRequest($request);
```

Comment:
you could have done the same thing as with create with the builder. it's ok to leave it like this for now tho